File: 283. Move Zeroes/283. Move Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int i = -1, j = 0, cnt = 0;
        while (++i < nums.size()) {
            if (nums[i] != 0) {
                nums[j++] = nums[i];
            } else {
                cnt++;
            }
        }
        while (cnt > 0) {
            nums[nums.size() - cnt--] = 0;
        }
        return;
    }
};

File: 283. Move Zeroes/283. Move Zeroes.java
public class Solution {
    public void moveZeroes(int[] nums) {
        int j = 0, cnt = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != 0) {
                nums[j++] = nums[i];
            } else {
                cnt++;
            }
        }
        if (cnt > 0){
            Arrays.fill(nums, nums.length - cnt, nums.length, 0);
        }
        return;
    }
}
